ecllibrary/std/OpenSSL.ecl

```diff
@@ -0,0 +1,355 @@
+     *                          passed in (e.g. (DATA)''); REQUIRED
+     * @param   iv              The IV to use during decryption; if not set
+     *                          then a random value will be used; if set,
+     *                          it must be of the expected size for the given
+     *                          algorithm; OPTIONAL, defaults to creating a
+     *                          random value
+     * @param   salt            The salt to use during decryption; if not set
+     *                          then a random value will be used; if set,
+     *                          it must be of the expected size for the given
+     *                          algorithm; OPTIONAL, defaults to creating a
+     *                          random value
+     *
+     * @return  The plaintext as a DATA type.
+     *
+     * @see     AvailableAlgorithms()
+     *          IVSize()
+     *          SaltSize()
+     *          Encrypt()
+     */
+    EXPORT DATA Decrypt(DATA ciphertext, VARSTRING algorithm_name, DATA passphrase, DATA iv = (DATA)'', DATA salt = (DATA)'') := lib_openssl.OpenSSL.cipherDecrypt(ciphertext, algorithm_name, passphrase, iv, salt);
+END; // Ciphers
+
+EXPORT PublicKey := MODULE
+
+    /**
+     * Perform a hybrid encryption using one or more RSA public keys.
+     *
+     * Because asymmetric encryption is computationally expensive, large
+     * payloads are actually encrypted with a symmetric cipher and a
+     * randomly-generated passphrase. The passphrase, which is much shorter,
+     * is then encrypted with the public key. The whole package is then bundled
+     * together into an "envelope" and "sealed".
+     *
+     * The function uses RSA public keys and they must be in PEM format. To
+     * generate such keys on the command line:
+     *
+     *      ssh-keygen -b 4096 -t rsa -m pem -f sample2
+     *      ssh-keygen -f sample2 -e -m pem > sample2.pub
+     *
+     * The resulting files, sample2 and sample2.pub, are the private and public
+     * keys, respectively. Their contents may be passed to this function.
+     *
+     * @param   plaintext           The data to encrypt; REQUIRED
+     * @param   pem_public_keys     One or more RSA public keys, in PEM format;
+     *                              note that this is a SET -- you can pass
+     *                              more than one public key here, and the resulting
+     *                              ciphertext can be decrypted by any one of the
+     *                              corresponding private keys; REQUIRED
+     * @param   algorithm_name      The name of the symmetric algorithm to use
+     *                              to encrypt the payload; must be one of those
+     *                              returned by Ciphers.AvailableAlgorithms();
+     *                              OPTIONAL, defaults to aes-256-cbc
+     *
+     * @return  The encrypted ciphertext.
+     *
+     * @see     Unseal()
```

Comment:
We did some renaming, so this Unseal() was changed to RSAUnseal(). Check all other @see functions as well.

Reply:
Fixed the @see functions in the comments.